.github/workflows/ci-test-coverage.yml

```diff
@@ -0,0 +1,51 @@
+name: ci-test-coverage
+
+on:
+  push:
+    branches: [main]
+    paths:
+      - "KubeArmor/**"
+      - "tests/**"
+      - "protobuf/**"
+      - ".github/workflows/ci-test-coverage.yml"
+  pull_request:
+    branches: [main]
+    paths:
+      - "KubeArmor/**"
+      - "tests/**"
+      - "protobuf/**"
+      - ".github/workflows/ci-test-coverage.yml"
+
+jobs:
+    codecoverage:
+    needs: [build]
+    - name: Measure code coverage / ${{ matrix.os }} / ${{ matrix.runtime }}
+    runs-on: ${{ matrix.os }}
+    env:
+      RUNTIME: ${{ matrix.runtime }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-20.04]
+        runtime: ["docker", "containerd", "crio"]
```

Comment:
coverage doesn't need a matrix run. 
Matrix runs helps us run across different envs to validate things. But we don't need it for this specific case since the sole work for this job is to download and upload artifacts.